core/src/test/scala/unit/kafka/log/LogSegmentTest.scala

```diff
@@ -143,6 +143,39 @@ class LogSegmentTest {
     checkEquals(ms2.records.iterator, read.records.records.iterator)
   }
 
+  @Test
+  def testReadWhenNoMaxPosition(): Unit = {
+    val maxPosition = -1
+    val maxSize = 1
+    val seg = createSegment(40)
+    val ms = records(50, "hello", "there")
+    seg.append(51, RecordBatch.NO_TIMESTAMP, -1L, ms)
+    for (minOneMessage <- Array(true, false)) {
+      // read before first offset
+      var read = seg.read(48, maxSize, maxPosition, minOneMessage)
```

Comment:
Hmm, maxSize=-1 is an invalid input, right?

Reply:
yes, when maxSize is 1, the following error will be thrown:

```
java.lang.IllegalArgumentException: Invalid max size -1 for log read from segment FileRecords(size=78, file=/var/folders/bq/w6tnkbq964q8sqpvj4fbmjr40000gq/T/kafka-18165220027855018994/00000000000000000040.log, start=0, end=2147483647)
	at org.apache.kafka.storage.internals.log.LogSegment.read(LogSegment.java:432)
```